### PR TITLE
Refactored updateTasksStatus.ts

### PR DESCRIPTION
**Pull request by endeavourmonk**
refactored code: removed if else ladder and used mapping for tasks status.

### CODE REVIEW

**Review by bhtibrewal — DISMISSED** [review]

**Review by bhtibrewal — COMMENTED** [review]
LGTM

File: src/helperFunctions/updateTasksStatus.ts [hunk 1]
```diff
@@ -2,18 +2,19 @@ import task from '../interfaces/task.type';
...
-        unassigned: 'AVAILABLE',
-        completed: 'COMPLETED',
-        blocked: 'BLOCKED',
-    };
+    enum Status {
+        active = 'IN_PROGRESS',
+        pending = 'IN_PROGRESS',
+        assigned = 'ASSIGNED',
+        unassigned = 'AVAILABLE',
+        completed = 'COMPLETED',
```
Comment (bhtibrewal) [inline src/helperFunctions/updateTasksStatus.ts:10]:
```
return tasks.map((taskItem) => {
       { ...taskItem, status :
            Status[taskList[index].status as keyof typeof Status]}
    });
```

**Review by bhtibrewal — CHANGES_REQUESTED** [review]

File: src/helperFunctions/updateTasksStatus.ts (hunk 1 above)
Comment (endeavourmonk) [reply src/helperFunctions/updateTasksStatus.ts:10]: @bhtibrewal updated please review

**Review by bhtibrewal — APPROVED** [review]
LGTM

**Review by RitikJaiswal75 — COMMENTED** [review]
I tried running your PR locally

This is what it shows

image

**Pull request closed without merging**